src/nvme/types.h

```diff
@@ -4696,6 +4718,162 @@ struct nvmf_discovery_log {
 	struct nvmf_disc_log_entry entries[];
 };
 
+/**
+ * Discovery Information Management (DIM) command. This is sent by a
+ * host to a Discovery Controller (DC) to perform explicit registration.
+ */
+#define NVMF_ENAME_LEN       256
+#define NVMF_EVER_LEN        64
+
+enum nvmf_dim_tas {
+	NVME_TAS_REGISTER   = 0x00,
+	NVME_TAS_DEREGISTER = 0x01,
+	NVME_TAS_UPDATE     = 0x02,
+};
+
+enum nvmf_disc_info_entfmt
+{
+	NVME_ENTFMT_BASIC    = 0x01,
+	NVME_ENTFMT_EXTENDED = 0x02,
+};
+
+enum nvmf_disc_info_etype
+{
+	NVME_REGISTRATION_HOST = 0x01,
+	NVME_REGISTRATION_DDC  = 0x02,
+	NVME_REGISTRATION_CDC  = 0x03,
+};
+
+enum nvmf_exattype
+{
+	NVME_EXATTYPE_HOSTID   = 0x01, /* Host Identifier */
+	NVME_EXATTYPE_SYMNAME  = 0x02, /* Symbolic Name */
+};
+
+/**
+ * struct nvmf_ext_attr - Extended Attribute (EXAT)
+ *
+ *              Bytes           Description
+ *   @type      01:00           Extended Attribute Type (EXATTYPE) - see enum nvmf_exattype
+ *   @len       03:02           Extended Attribute Length (EXATLEN)
+ *   @val       (EXATLEN-1)     Extended Attribute Value (EXATVAL) - size allocated for array must be a multiple of 4 bytes
+ *              +4:04
+ */
+struct nvmf_ext_attr
+{                                                           /* Bytes */
+	__le16			type;                       /* 01:00 */
```

Comment:
and given those things are already documented above do we really need those infos again?